tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/BatchData.java

```diff
@@ -107,16 +93,27 @@ public boolean isEmpty() {
   }
 
   public boolean hasCurrent() {
-    return curIdx < count;
+    if (readCurListIndex < writeCurListIndex) {
+      return readCurArrayIndex < capacity;
+    }
+    else if (readCurListIndex == writeCurListIndex) {
+      return readCurArrayIndex < writeCurArrayIndex;
```

Comment:
should this be <=?

This method could be implemented as :   return readCurListIndex <= writeCurListIndex && readCurArrayIndex <= writeCurArrayIndex;